ui/platform_window/x11/x11_window_base.cc

```diff
@@ -534,17 +536,38 @@ void X11WindowBase::OnWMStateUpdated() {
   // Propagate the window minimization information to the client.
   ui::PlatformWindowState state =
       ui::PlatformWindowState::PLATFORM_WINDOW_STATE_NORMAL;
-  if (IsMinimized() != was_minimized) {
-    if (IsMinimized()) {
-      state = ui::PlatformWindowState::PLATFORM_WINDOW_STATE_MINIMIZED;
-    } else if (IsMaximized()) {
-      // When the window is recovered from minimized state, set state to the
-      // previous maximized state if it was like that. Otherwise, NORMAL state
-      // will be set.
-      state = ui::PlatformWindowState::PLATFORM_WINDOW_STATE_MAXIMIZED;
-    }
+  if (IsMaximized())
+    state = ui::PlatformWindowState::PLATFORM_WINDOW_STATE_MAXIMIZED;
+  else if (IsMinimized())
+    state = ui::PlatformWindowState::PLATFORM_WINDOW_STATE_MINIMIZED;
+
+  // |restored_bounds_in_pixels_| can tell if the maximize/restore has been
+  // triggered by client or not. Typically, if it was the client who triggered
+  // that, |restored_bounds_in_pixels_| had values stored. Thus, we can be
+  // sure the client is not flooded with window states when the sources of a
+  // change has been the client itself.
+  if ((restored_bounds_in_pixels_.IsEmpty() &&
+       IsMaximized() != was_maximized) ||
+      IsMinimized() != was_minimized) {
     delegate_->OnWindowStateChanged(state);
+    // A workaround to update the opaque view layout:
+    // In a case when the state of a window has been changed due to some server
```

Comment:
I fear from "server side events" not being a good term, from an Ozone perspective. maybe use "react to host window manager event" instead?

Reply:
done